openqa-investigate

```diff
@@ -1,391 +1,178 @@
-        return 0
-    fi
-
-    # determine dependency data or postpone if cluster not done
-    dependency_data=$(query_dependency_data_or_postpone "$id" "$job_data") || rc=$?
-    [[ $rc != 255 ]] && return $rc
-
-    # determine the job in the cluster with the lowest ID to use that for commenting/synchronization
-    first_cluster_job_id=$(echo "$dependency_data" | runjq -r "[$id, [.cluster[] | select(contains([$id]))]] | flatten | min") || return $?
-
-    [[ "$old_name" =~ $exclude_name_regex ]] && echo "Job name '$old_name' matches \$exclude_name_regex '$exclude_name_regex', skipping investigation" && return 0
-    group="$(echo "$job_data" | runjq -r '.job.parent_group + " / " + .job.group')" || return $?
-    [[ "$group" = " / " ]] && [[ "$exclude_no_group" = "true" ]] && echo "Job w/o job group, \$exclude_no_group is set, skipping investigation" && return 0
-    [[ "$group" =~ $exclude_group_regex ]] && echo "Job group '$group' matches \$exclude_group_regex '$exclude_group_regex', skipping investigation" && return 0
-
-    # Optionally we can find "first failed", could extend openQA investigation
-    # method instead for we are just working based on supplied job which can
-    # have more, ambiguous potential changes that we need to bisect on
-
-    # sync by writing initial investigation comment (edited later)
-    rc=0
-    comment_id=$(sync_via_investigation_comment "$id" "$first_cluster_job_id") || rc=$?
-    [[ $rc != 255 ]] && return $rc
-
-    out=$(trigger_jobs "$id" "${@:2}")
-    $verbose && echo "$0, id: '$id', out: '$out'"
-    finalize_investigation_comment "$id" "$first_cluster_job_id" "$comment_id" "$out"
-}
-
-main() {
-    local id=${1:?"Need 'job_id'"}
-    client_prefix=''
-    [ "$dry_run" = "1" ] && client_prefix="echo"
-    set +u
-    if [[ -z "${client_call[*]}" ]]; then
-        client_call=(openqa-cli "${client_args[@]}")
-        client_prefix="${client_prefix:-runcli}"
-        client_call=("$client_prefix" "${client_call[@]}")
-    fi
-    set -u
-    clone_call="${clone_call:-"$client_prefix openqa-clone-job --json-output --skip-chained-deps --max-depth 0 --parental-inheritance --within-instance"}"
-    investigate "$@"
-}
-
-caller 0 > /dev/null || main "$@"
+import requests
+import json
+import argparse
+import os
+import re
+import time
+import subprocess
+
+errorlog = ""
+
+def warn(message):
+    print(message, file=sys.stderr)
+
+def runcli(args):
+    try:
+        result = subprocess.run(args, capture_output=True, text=True, check=True)
+        return result.stdout
+    except subprocess.CalledProcessError as e:
+        warn(f"Command {args} failed with error: {e.stderr}")
+        return None
+
+def runjq(input_data, jq_filter):
+    try:
+        result = subprocess.run(['jq', jq_filter], input=input_data, capture_output=True, text=True, check=True)
+        return result.stdout
+    except subprocess.CalledProcessError as e:
+        warn(f"jq error: {e.stderr}")
```

Comment:
Will we actually get the original input here? That's what the bash function is doing currently. If jq fails we want to know what the input was.

Reply:
-> https://github.com/os-autoinst/scripts/pull/332?notification_referrer_id=NT_kwDOABnW-LMxMTYzMjQ3OTAzMjoxNjkzNDMy&notifications_query=is%3Aunread#discussion_r1689552634